### PR TITLE
Update gitAndTools.gita and add bash completion

### PR DESCRIPTION
**Pull request by wigust**
This package updates gitAndTools.gita and installs bash completion.

###### Motivation for this change

Missing bash completion for gita.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by seqizz — APPROVED** [review]
Looks like the completion file is not in release tarballs. LGTM

File: pkgs/applications/version-management/git-and-tools/gita/default.nix [hunk 1]
```diff
@@ -1,24 +1,38 @@
...
   propagatedBuildInputs = [
     pyyaml
     setuptools
   ];
 
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
+    installShellCompletion --bash --name gita ${
+      fetchurl {
```
Comment (marsam) [inline pkgs/applications/version-management/git-and-tools/gita/default.nix:28]: looks like the completions are part of the repository, but aren't in the pypi tarball. 
Would you mind replacing `fetchPypi` with `fetchFromGitHub`? and install the completions from there

**Review by seqizz — APPROVED** [review]
Tested again, LGTM.

File: pkgs/applications/version-management/git-and-tools/gita/default.nix (hunk 1 above)
Comment (wigust) [reply pkgs/applications/version-management/git-and-tools/gita/default.nix]: Done.  Also I enabled tests if you don't mind.

**Review by seqizz — APPROVED** [review]

**Comment by marsam** [comment]
Sorry for the delay and thank you!

**Pull request merged**